File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for c in s:
        if c == '(' or c == '[':
                stack.append(c)
        if not stack:
            if c == ')' or c == ']':
                return False
        if stack:
            if c == ')':
                if stack.pop() == '(':
                    continue
                else:
                    return False
            if c == ']':
                if stack.pop() == '[':
                    continue
                else:
                    return False
    if stack:
        return False
    if not stack:
        return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    Stack<Character> stack = new Stack<>();
    char c;
    
    boolean solution(String s) {
        boolean answer = true;
        
        for(int i=0; i<s.length(); i++) {
            c = s.charAt(i);
            if(c == '(') {
                stack.push(c);
            }
            if(c == ')') {
                if(stack.isEmpty()) {
                    return false;
                }
                else {
                    if(stack.peek() == '(') {
                        stack.pop();
                    }
                    else {
                        return false;
                    }
                }
            }
        }
        
        if(!stack.isEmpty()) {
            return false;
        }
        else {
            return true;
        }
    }
}